test/mock_batch_test.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:cloud_firestore_mocks/cloud_firestore_mocks.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() async {
+  TestWidgetsFlutterBinding.ensureInitialized();
+  String users = "users";
+  MockFirestoreInstance firestore = MockFirestoreInstance();
+
+  Map<String, dynamic> userDocumentMock(String userId) {
+    Map<String, dynamic> data = {
+      "id": userId,
+      "updatedAt": DateTime.now(),
+      "createdAt": DateTime.now(),
+    };
+    return data;
+  }
+
+  Future<List<Map<String, dynamic>>> createUsers({int times}) async {
+    List<Map<String, dynamic>> userDataList = [];
+    for (int i = 1; i <= times; ++i) {
+      String userId = i.toString();
+      userDataList.add(userDocumentMock(userId));
+    }
+    return userDataList;
+  }
+
+  setUp(() async {});
+  tearDown(() async {});
+
+  group("batch", () {
+    test("succees when create 500 documents at once.", () async {
+      List<Map<String, dynamic>> userDataList = await createUsers(times: 500);
+      int userDataListLength = userDataList.length;
+      for (int i = 0; i < userDataListLength; i = i + 500) {
+        int first = i;
+        int last = i + 500;
+        if (last >= userDataListLength) {
+          last = userDataListLength;
+        }
+        var selectUserDataList = userDataList.getRange(first, last);
+        var batch = firestore.batch();
+        await Future.forEach(selectUserDataList, (data) async {
```

Comment:
`Firestore.collection``CollectionRef.doc` and `BatchWrite.set` are all synchronous, so you don't need Futures.

Also `"${data["id"]}"` can be simplified to `data["id"]`. Both are strings.

As a result, you can rewrite this as:

```dart
for (final userData in selectUserDataList) {
  final documentRef = firestore.collection(users).doc(userData["id"]);
  batch.set(documentRef, userData);
}
```